netsim/providers/__init__.py

```diff
@@ -131,63 +138,104 @@ def create_extra_files_mappings(
     map_dict = filemaps.mapping_to_dict(mappings)
     cur_binds = node.get(f'{self.provider}.{outkey}',[])
     bind_dict = filemaps.mapping_to_dict(cur_binds)
+    base_path = pathlib.Path(f"{self.provider}_files")
+    
     for file,mapping in map_dict.items():
       file = file.replace('@','.')
-      if file in bind_dict:
+      # Check if mapping ends with :shared - if so, put it in the root provider folder
+      if mapping.endswith(self.SHARED_SUFFIX):
+        mapping = mapping.rsplit(self.SHARED_SUFFIX, 1)[0] + self.READ_ONLY_SUFFIX
+        # Prefix shared files to avoid node name conflicts
+        out_path = base_path / f"{self.SHARED_PREFIX}{file}" 
+      else:
+        out_path = base_path / node.name / file
+
+      out_key = out_path.as_posix()
+      if out_key in bind_dict:
         continue
+
       if not self.find_extra_template(node,file,topology):
         log.error(
           f"Cannot find template {file}.j2 for extra file {self.provider}.{inkey}.{file} on node {node.name}",
           category=log.IncorrectValue,
           module=self.provider)
         continue
 
-      out_folder = f"{self.provider}_files/{node.name}"
-      bind_dict[f"{out_folder}/{file}"] = mapping         # note: node_files directory is flat
+      bind_dict[out_key] = mapping
 
     node[self.provider][outkey] = filemaps.dict_to_mapping(bind_dict)
 
   def create_extra_files(
       self,
       node: Box,
       topology: Box,
-      inkey: str = 'config_templates',
       outkey: str = 'binds') -> None:
 
     binds = node.get(f'{self.provider}.{outkey}',None)
     if not binds:
       return
 
     sys_folder = str(_files.get_moddir())+"/"
-    out_folder = f"{self.provider}_files/{node.name}"
 
     bind_dict = filemaps.mapping_to_dict(binds)
-    for file,mapping in bind_dict.items():
-      if not out_folder in file:                  # Skip files that are not mapped into the temporary provider folder
+    
+    node_data = {
+        **node.to_dict(),
+        'hostvars': topology.nodes.to_dict(),
+        'hosts': get_host_addresses(topology),
+        'addressing': topology.addressing.to_dict()
+    }
+
+    base_path = pathlib.Path(f"{self.provider}_files")
+
+    for file_str,mapping in bind_dict.items():
+      full_out_path = pathlib.Path(file_str)
+
+      # Only handle files placed under the provider's file directory
+      try:
+        full_out_path.relative_to(base_path)
+      except ValueError:
+        continue
+
+      # The template name is the file name part of the path
+      file_rel = full_out_path.name
```

Comment:
Showstopper: this breaks the ability to have relative paths in template names, for example:

```
nodes:
  h1:
    clab.config_templates:
      _n_r1: /etc/nr1
      a/_n_r2: /etc/nr2
```